File: src/main/java/pl/yalgrin/playnite/simplesync/repository/TagRepository.java
package pl.yalgrin.playnite.simplesync.repository;

import org.springframework.data.r2dbc.repository.Query;
import org.springframework.stereotype.Repository;
import pl.yalgrin.playnite.simplesync.domain.Tag;
import reactor.core.publisher.Flux;

@Repository
public interface TagRepository extends ObjectRepository<Tag> {
    @Query("select c.id from playnite_tag c order by c.id asc")
    Flux<Long> findAllIds();
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/library/repository/TagRepository.kt
package pl.yalgrin.playnite.simplesync.library.repository

import org.springframework.data.r2dbc.repository.Query
import org.springframework.stereotype.Repository
import pl.yalgrin.playnite.simplesync.library.domain.Tag
import reactor.core.publisher.Flux

@Repository
interface TagRepository : ObjectRepository<Tag> {
    @Query("select c.id from playnite_tag c order by c.id")
    override fun findAllIds(): Flux<Long>
}